components/board/pi/impl/board.go

```diff
@@ -164,20 +164,23 @@ func (pi *piPigpio) Reconfigure(
 	}
 
 	pi.mu.Lock()
-	defer pi.mu.Unlock()
 
 	if err := pi.reconfigureI2cs(ctx, cfg); err != nil {
+		pi.mu.Unlock()
```

Comment:
Yuck! What's this all for!? It's going to be so easy to write bugs this way. Someone is going to forget to unlock the mutex in one of these, or will copy and paste a paragraph similar to this down below the `defer` statement, and it'll be a real pain to remember how to do this right.

Reply:
You're right, your suggestion is way better